.github/workflows/build.yaml

```diff
@@ -0,0 +1,36 @@
+name: CI
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  
+jobs:
+  test-fedora:
+    name: fedora
+    runs-on: ubuntu-latest
```

Comment:
Isn't this testing on fedora?

Reply:
Yes, this tests on Fedora. Hope that is ok.  It is possible to add other operating systems later.

Reply:
It runs a Fedora container within Ubuntu.

Reply:
Don't think we need container within the job. Lets keep it simple: get rid of podman and just build it here. (I don't have a preference between fedora or ubuntu, either is okay). But do build with g++ >= 12.x.

Reply:
The container may be helpful to run on Linux Os other than Ubuntu. For example this would also help run [Clear Linux](https://clearlinux.org/downloads/containers). Can define jobs for Ubuntu, Clear Linux and Fedora.  A test for Fedora is helpful as it is being packaged for Fedora.